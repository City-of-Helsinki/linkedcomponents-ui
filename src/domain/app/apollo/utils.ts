import { EMPTY_MULTI_LANGUAGE_OBJECT } from '../../../constants';
import {
  DataSource,
  Division,
  Enrolment,
  Event,
  ExternalLink,
  Image,
  Keyword,
  KeywordSet,
  Language,
  LocalisedObject,
  Meta,
  Offer,
  Organization,
  OrganizationClass,
  Place,
  Position,
  Registration,
  User,
} from '../../../generated/graphql';
import getValue from '../../../utils/getValue';

export const addTypenameDataSource = (
  dataSource?: DataSource | null
): DataSource | null =>
  dataSource
    ? {
        ...dataSource,
        __typename: 'DataSource',
      }
    : null;

export const addTypenameDivision = (
  division?: Division | null
): Division | null =>
  division
    ? {
        ...division,
        name: addTypenameLocalisedObject(division.name),
        __typename: 'Division',
      }
    : null;

export const addTypenameEnrolment = (
  enrolment?: Enrolment | null
): Enrolment | null =>
  enrolment
    ? {
        ...enrolment,
        __typename: 'Enrolment',
      }
    : null;

export const addTypenameEvent = (event?: Event | null): Event | null =>
  event
    ? {
        ...event,
        description: addTypenameLocalisedObject(event.description),
        infoUrl: addTypenameLocalisedObject(event.infoUrl),
        name: addTypenameLocalisedObject(event.name),
        provider: addTypenameLocalisedObject(event.provider),
        shortDescription: addTypenameLocalisedObject(event.shortDescription),
        externalLinks: getValue(event.externalLinks, [])?.map((externalLink) =>
          addTypenameExternalLink(externalLink)
        ),
        images: getValue(
          event.images?.map((image) => addTypenameImage(image)),
          []
        ),
        inLanguage: getValue(
          event.inLanguage?.map((language) => addTypenameLanguage(language)),
          []
        ),
        keywords: getValue(
          event.keywords?.map((keyword) => addTypenameKeyword(keyword)),
          []
        ),
        audience: getValue(
          event.audience?.map((keyword) => addTypenameKeyword(keyword)),
          []
        ),
        location: addTypenamePlace(event.location),
        offers: getValue(
          event.offers?.map((offer) => addTypenameOffer(offer)),
          []
        ),
        superEvent: event.superEvent
          ? addTypenameEvent(event.superEvent)
          : null,
        subEvents: getValue(
          event.subEvents?.map((subEvent) => addTypenameEvent(subEvent)),
          []
        ),
        __typename: 'Event',
      }
    : null;

export const addTypenameExternalLink = (
  externalLink?: ExternalLink | null
): ExternalLink | null =>
  externalLink ? { ...externalLink, __typename: 'ExternalLink' } : null;

export const addTypenameImage = (image?: Image | null): Image | null =>
  image
    ? {
        ...image,
        altText: addTypenameLocalisedObject(image.altText),
        __typename: 'Image',
      }
    : null;

export const addTypenameKeyword = (keyword?: Keyword | null): Keyword | null =>
  keyword
    ? {
        ...keyword,
        name: addTypenameLocalisedObject(keyword?.name),
        __typename: 'Keyword',
      }
    : null;

export const addTypenameKeywordSet = (
  keywordSet?: KeywordSet | null
): KeywordSet | null =>
  keywordSet
    ? {
        ...keywordSet,
        keywords: getValue(
          keywordSet.keywords?.map((keyword) => addTypenameKeyword(keyword)),
          []
        ),
        name: addTypenameLocalisedObject(keywordSet?.name),
        __typename: 'KeywordSet',
      }
    : null;

export const addTypenameLanguage = (
  language?: Language | null
): Language | null =>
  language
    ? {
        ...language,
        name: addTypenameLocalisedObject(language?.name),
        __typename: 'Language',
      }
    : null;

export const addTypenameLocalisedObject = (
  item?: string | LocalisedObject | null
): LocalisedObject | null =>
  item
    ? {
        ...(typeof item === 'string'
          ? { ...EMPTY_MULTI_LANGUAGE_OBJECT, fi: item }
          : item),
        __typename: 'LocalisedObject',
      }
    : null;

export const addTypenameMeta = (meta: Meta): Meta => ({
  ...meta,
  __typename: 'Meta',
});

export const addTypenameOffer = (offer?: Offer | null): Offer | null =>
  offer
    ? {
        ...offer,
        description: addTypenameLocalisedObject(offer.description),
        infoUrl: addTypenameLocalisedObject(offer.infoUrl),
        price: addTypenameLocalisedObject(offer.price),
        __typename: 'Offer',
      }
    : null;

export const addTypenameOrganization = (
  organization?: Organization | null
): Organization | null =>
  organization ? { ...organization, __typename: 'Organization' } : null;

export const addTypenameOrganizationClass = (
  organizationClass?: OrganizationClass | null
): OrganizationClass | null =>
  organizationClass
    ? { ...organizationClass, __typename: 'OrganizationClass' }
    : null;

export const addTypenamePlace = (place?: Place | null): Place | null =>
  place
    ? {
        ...place,
        addressLocality: addTypenameLocalisedObject(place.addressLocality),
        divisions: getValue(
          place.divisions?.map((division) => addTypenameDivision(division)),
          []
        ),
        infoUrl: addTypenameLocalisedObject(place.infoUrl),
        name: addTypenameLocalisedObject(place.name),
        position: addTypenamePosition(place.position),
        streetAddress: addTypenameLocalisedObject(place.streetAddress),
        telephone: addTypenameLocalisedObject(place.telephone),
        __typename: 'Place',
      }
    : null;

export const addTypenamePosition = (
  position?: Position | null
): Position | null =>
  position ? { ...position, __typename: 'Position' } : null;

export const addTypenameRegistration = (
  registration?: Registration | null
): Registration | null =>
  registration
    ? {
        ...registration,
        event: addTypenameEvent(registration?.event),
        signups: Array.isArray(registration.signups)
          ? registration.signups.map((enrolment) =>
              addTypenameEnrolment(enrolment)
            )
          : [],
        __typename: 'Registration',
      }
    : null;

export const addTypenameUser = (user?: User | null): User | null =>
  user ? { ...user, __typename: 'User' } : null;
