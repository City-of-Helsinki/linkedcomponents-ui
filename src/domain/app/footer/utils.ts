import { ROUTES } from '../../../constants';
import { UserFieldsFragment } from '../../../generated/graphql';
import { areRegistrationRoutesAllowed } from '../../user/permissions';
import { NavigationGroup, NavigationGroupItem } from './types';

const getNavigationGroupItemsInstructions = (
  user?: UserFieldsFragment
): NavigationGroupItem[] => [
  {
    label: 'helpPage.sideNavigation.labelEventsInstructions',
    url: ROUTES.INSTRUCTIONS_EVENTS,
  },
  ...(areRegistrationRoutesAllowed(user)
    ? [
        {
          label: 'helpPage.sideNavigation.labelRegistrationInstructions',
          url: ROUTES.INSTRUCTIONS_REGISTRATION,
        },
      ]
    : []),
  {
    label: 'helpPage.sideNavigation.labelFaq',
    url: ROUTES.INSTRUCTIONS_FAQ,
  },
];

export const getNavigationGroupInstructions = (
  user?: UserFieldsFragment
): NavigationGroup => ({
  heading: 'helpPage.sideNavigation.labelInstructions',
  headingLink: ROUTES.INSTRUCTIONS,
  items: getNavigationGroupItemsInstructions(user),
});
