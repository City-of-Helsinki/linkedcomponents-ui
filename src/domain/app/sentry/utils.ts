import * as Sentry from '@sentry/browser';
import * as H from 'history';

import { UserFieldsFragment } from '../../../generated/graphql';

const reportError = ({
  data,
  location,
  message,
  user,
}: {
  data: { error: Record<string, unknown> } & Record<string, unknown>;
  location: H.Location;
  message: string;
  user?: UserFieldsFragment;
}): string => {
  const { error, ...restData } = data;
  const reportObject = {
    extra: {
      data: {
        ...restData,
        currentUrl: window.location.href,
        location,
        timestamp: new Date(),
        user,
        userAsString: user ? JSON.stringify(user) : '',
        userAgent: navigator.userAgent,
        error,
        errorAsString: JSON.stringify(error),
      },
      level: 'error',
    },
  };

  return Sentry.captureException(message, reportObject);
};

export { reportError };
