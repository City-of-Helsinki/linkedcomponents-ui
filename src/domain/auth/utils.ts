/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-explicit-any */
import * as Sentry from '@sentry/react';
import axios, { AxiosResponse } from 'axios';
import { TFunction } from 'i18next';
import { User, UserManager, UserManagerSettings } from 'oidc-client';
import React from 'react';
import { toast } from 'react-toastify';

import { Language } from '../../types';
import getUnixTime from '../../utils/getUnixTime';
import getValue from '../../utils/getValue';
import {
  API_SCOPE,
  API_TOKEN_EXPIRATION_TIME,
  API_TOKEN_NOTIFICATION_TIME,
  ApiTokenActionTypes,
  OidcActionTypes,
} from './constants';
import {
  ApiTokenAction,
  ApiTokenReducerState,
  OidcAction,
  OidcReducerState,
} from './types';

const apiAccessTokenStorage = sessionStorage;

const storageKey = `oidc.apiToken.${API_SCOPE}`;

const MAINTENANCE_MODE =
  process.env.REACT_APP_ENABLE_MAINTENANCE_MODE === 'true';

export const getApiTokenFromStorage = (): string | null =>
  apiAccessTokenStorage.getItem(storageKey);

export const setApiTokenToStorage = (accessToken: string): void =>
  apiAccessTokenStorage.setItem(storageKey, accessToken);

export const clearApiTokenFromStorage = (): void =>
  apiAccessTokenStorage.removeItem(storageKey);

export const createUserManager = (config: UserManagerSettings) => {
  return new UserManager(config);
};

// Oidc auth actions
// event callback when the access token expired
export const onAccessTokenExpired = ({
  dispatchOidcState,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
}) => {
  dispatchOidcState({ type: OidcActionTypes.USER_EXPIRED, payload: null });
};

// event callback when the user is expiring
export const onAccessTokenExpiring = ({
  dispatchOidcState,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
}) => {
  dispatchOidcState({ type: OidcActionTypes.USER_EXPIRING, payload: null });
};

// event callback when silent renew errored
export const onSilentRenewError = ({
  dispatchOidcState,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
}) => {
  dispatchOidcState({
    type: OidcActionTypes.SILENT_RENEW_ERROR,
    payload: null,
  });
};

// event callback when the user has been loaded (on silent renew or redirect)
export const onUserLoaded = ({
  dispatchOidcState,
  user,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
  user: User;
}) => {
  dispatchOidcState({ type: OidcActionTypes.USER_FOUND, payload: user });
};

// event callback when the user is signed out
export const onUserSignedOut = ({
  dispatchOidcState,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
}) => {
  dispatchOidcState({ type: OidcActionTypes.USER_SIGNED_OUT, payload: null });
};

// event callback when the user is logged out
export const onUserUnloaded = ({
  dispatchOidcState,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
}) => {
  dispatchOidcState({
    type: OidcActionTypes.SESSION_TERMINATED,
    payload: null,
  });
};

// callback function called when the user has been loaded
export const getUserCallback = ({
  dispatchOidcState,
  user,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
  user: User | null;
}) => {
  /* istanbul ignore else */
  if (user && !user.expired) {
    onUserLoaded({ dispatchOidcState, user });
  } else if (!user || (user && user.expired)) {
    onAccessTokenExpired({ dispatchOidcState });
  }
  return user;
};

// error callback called when the userManager's loadUser() function failed
export const errorCallback = ({
  dispatchOidcState,
  error,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
  error: Error;
}) => {
  console.error(`AuthContext: Error in loadUser() function: ${error.message}`);
  dispatchOidcState({ type: OidcActionTypes.LOAD_USER_ERROR, payload: null });
};

// function to load the current user into the store
// NOTE: use only when silent renew is configured
export const loadUser = async ({
  dispatchOidcState,
  userManager,
}: {
  dispatchOidcState: React.Dispatch<OidcAction>;
  userManager: UserManager;
}): Promise<void> => {
  dispatchOidcState({ type: OidcActionTypes.LOADING_USER, payload: null });

  userManager
    .getUser()
    .then((user) => getUserCallback({ dispatchOidcState, user }))
    .catch((error) => errorCallback({ dispatchOidcState, error }));
};

export const signIn = async ({
  locale,
  path,
  t,
  userManager,
}: {
  locale: Language;
  path?: string;
  t: TFunction;
  userManager: UserManager;
}): Promise<void> => {
  if (MAINTENANCE_MODE) {
    toast.error(getValue(t('maintenance.toast'), ''));

    return Promise.reject();
  }

  await userManager
    .signinRedirect({
      data: { path: path || '/' },
      ui_locales: locale,
    })
    .catch((error) => {
      if (error.message === 'Network Error') {
        toast.error(getValue(t('authentication.networkError.message'), ''));
      } else {
        toast.error(getValue(t('authentication.errorMessage'), ''));
        Sentry.captureException(error);
      }
    });
};

export const signOut = async ({
  resetApiTokenData,
  userManager,
}: {
  resetApiTokenData: () => Promise<void>;
  userManager: UserManager;
}): Promise<void> => {
  try {
    await clearAllState(resetApiTokenData);
    await userManager.signoutRedirect();
  } catch (e) /* istanbul ignore next */ {
    Sentry.captureException(e);
  }
};

export const clearAllState = async (
  resetApiTokenData: () => Promise<void>
): Promise<void> => {
  await Promise.all([
    // Clear backend auth data
    await resetApiTokenData(),
    // Clear session storage
    sessionStorage.clear(),
  ]);
};

// Api token action
export const startFetchingToken = ({
  dispatchApiTokenState,
}: {
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
}) => {
  dispatchApiTokenState({
    type: ApiTokenActionTypes.START_FETCHING_TOKEN,
    payload: null,
  });
};

export const fetchTokenSuccess = async ({
  apiToken,
  dispatchApiTokenState,
}: {
  apiToken: string;
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
}) => {
  setApiTokenToStorage(apiToken);
  dispatchApiTokenState({
    type: ApiTokenActionTypes.FETCH_TOKEN_SUCCESS,
    payload: apiToken,
  });
};

export const fetchTokenError = ({
  dispatchApiTokenState,
  error,
}: {
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
  error: Error;
}) => {
  clearApiTokenFromStorage();
  dispatchApiTokenState({
    type: ApiTokenActionTypes.FETCH_TOKEN_ERROR,
    payload: error,
  });
};

export const renewApiToken = async ({
  accessToken,
  dispatchApiTokenState,
  t,
}: {
  accessToken: string;
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
  t: TFunction;
}) => {
  const apiTokensUrl = process.env.REACT_APP_OIDC_API_TOKENS_URL;

  if (!apiTokensUrl) {
    throw new Error(
      'Application configuration error, missing Tunnistamo api tokens url.'
    );
  }

  try {
    const res: AxiosResponse = await axios.get(apiTokensUrl, {
      headers: { Authorization: `bearer ${accessToken}` },
    });

    const apiToken = res.data[API_SCOPE];
    fetchTokenSuccess({ apiToken, dispatchApiTokenState });
  } catch (error: any) {
    fetchTokenError({ error, dispatchApiTokenState });
    toast.error(getValue(t('authentication.errorMessage'), ''));
  }
};

export const getApiToken = async ({
  accessToken,
  dispatchApiTokenState,
  t,
}: {
  accessToken: string;
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
  t: TFunction;
}) => {
  startFetchingToken({ dispatchApiTokenState });
  await renewApiToken({ accessToken, dispatchApiTokenState, t });
};

export const resetApiTokenData = ({
  dispatchApiTokenState,
}: {
  dispatchApiTokenState: React.Dispatch<ApiTokenAction>;
}) => {
  clearApiTokenFromStorage();
  dispatchApiTokenState({
    type: ApiTokenActionTypes.RESET_API_TOKEN_DATA,
    payload: null,
  });
};

export const getApiTokenExpirationTime = (): number =>
  getUnixTime(new Date()) + API_TOKEN_EXPIRATION_TIME;

export const isApiTokenExpiring = (expirationTime: number | null): boolean =>
  Boolean(
    expirationTime &&
      getUnixTime(new Date()) > expirationTime - API_TOKEN_NOTIFICATION_TIME
  );

export const getIsAuthenticated = ({
  apiTokenState,
  oidcState,
}: {
  apiTokenState: ApiTokenReducerState;
  oidcState: OidcReducerState;
}) => !!oidcState.user && !!apiTokenState.apiToken;

export const getIsLoading = ({
  apiTokenState,
  oidcState,
}: {
  apiTokenState: ApiTokenReducerState;
  oidcState: OidcReducerState;
}) => oidcState.isLoadingUser || apiTokenState.isLoadingApiToken;
