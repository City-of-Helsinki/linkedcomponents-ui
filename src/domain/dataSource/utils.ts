import {
  DataSourceFieldsFragment,
  DataSourceQueryVariables,
  DataSourcesQueryVariables,
} from '../../generated/graphql';
import { PathBuilderProps } from '../../types';
import getValue from '../../utils/getValue';
import queryBuilder from '../../utils/queryBuilder';
import { DataSourceFields } from './types';

export const dataSourcePathBuilder = ({
  args,
}: PathBuilderProps<DataSourceQueryVariables>): string => {
  const { id } = args;

  return `/data_source/${id}/`;
};

export const dataSourcesPathBuilder = ({
  args,
}: PathBuilderProps<DataSourcesQueryVariables>): string => {
  const { page, pageSize } = args;

  const variableToKeyItems = [
    { key: 'page', value: page },
    { key: 'page_size', value: pageSize },
  ];

  const query = queryBuilder(variableToKeyItems);

  return `/data_source/${query}`;
};

export const getDataSourceFields = (
  dataSource: DataSourceFieldsFragment
): DataSourceFields => {
  const id = getValue(dataSource.id, '');

  return {
    id,
    name: getValue(dataSource.name, ''),
  };
};
