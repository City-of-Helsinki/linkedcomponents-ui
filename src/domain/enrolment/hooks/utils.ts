import { TFunction } from 'i18next';

import { LEServerError, ServerErrorItem } from '../../../types';
import parseServerErrorMessage from '../../../utils/parseServerErrorMessage';
import pascalCase from '../../../utils/pascalCase';

export const parseEnrolmentServerErrors = ({
  result,
  t,
}: {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  result: LEServerError;
  t: TFunction;
}): ServerErrorItem[] => {
  // LE returns errors as array when trying to create/edit multiple enrolments in same request.
  // In that case call parseEnrolmentServerErrors recursively to get all single errors
  if (Array.isArray(result)) {
    return result.reduce(
      (previous: ServerErrorItem[], r) => [
        ...previous,
        ...parseEnrolmentServerErrors({ result: r, t }),
      ],
      []
    );
  }

  return typeof result === 'string'
    ? [{ label: '', message: parseServerErrorMessage({ error: [result], t }) }]
    : Object.entries(result).reduce(
        (previous: ServerErrorItem[], [key, error]) => [
          ...previous,
          ...parseEnrolmentServerError({ error: error as LEServerError, key }),
        ],
        []
      );

  // Get error item for an single error.
  function parseEnrolmentServerError({
    error,
    key,
  }: {
    error: LEServerError;
    key: string;
  }) {
    switch (key) {
      case 'detail':
      case 'non_field_errors':
        return [{ label: '', message: parseServerErrorMessage({ error, t }) }];
      default:
        return [
          {
            label: parseEnrolmentServerErrorLabel({ key }),
            message: parseServerErrorMessage({ error, t }),
          },
        ];
    }
  }

  // Get correct field name for an error item
  function parseEnrolmentServerErrorLabel({ key }: { key: string }): string {
    return t(`enrolment.form.label${pascalCase(key)}`);
  }
};
