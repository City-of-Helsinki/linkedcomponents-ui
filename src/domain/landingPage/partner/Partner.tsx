import React from 'react';

import styles from './partner.module.scss';

type Props = {
  href: string;
  imageUrl: string;
  name: string;
} & React.AnchorHTMLAttributes<HTMLAnchorElement>;

const Partner: React.FC<Props> = ({ href, imageUrl, name, ...rest }) => {
  return (
    <a {...rest} href={href} className={styles.partner}>
      <img src={imageUrl} alt={name} />
    </a>
  );
};

export default Partner;
