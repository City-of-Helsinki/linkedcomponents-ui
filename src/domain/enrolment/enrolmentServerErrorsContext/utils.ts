import { TFunction } from 'i18next';

import { LEServerError, ServerErrorItem } from '../../../types';
import isGenericServerError from '../../../utils/isGenericServerError';
import parseServerErrorMessage from '../../../utils/parseServerErrorMessage';
import { parseServerErrors } from '../../../utils/parseServerErrors';
import pascalCase from '../../../utils/pascalCase';

export const parseEnrolmentServerErrors = ({
  result,
  t,
}: {
  result: LEServerError;
  t: TFunction;
}): ServerErrorItem[] => {
  return parseServerErrors({
    parseServerError: parseEnrolmentServerError,
    result,
    t,
  });

  // Get error item for an single error.
  function parseEnrolmentServerError({
    error,
    key,
  }: {
    error: LEServerError;
    key: string;
  }) {
    if (key === 'signups') {
      return parseSignupServerError(error);
    }

    return [
      {
        label: parseEnrolmentServerErrorLabel({ key }),
        message: parseServerErrorMessage({ error, t }),
      },
    ];
  }

  // Get error items for video fields
  function parseSignupServerError(error: LEServerError): ServerErrorItem[] {
    /* istanbul ignore else */
    if (Array.isArray(error)) {
      return Object.entries(error[0]).reduce(
        (previous: ServerErrorItem[], [key, e]) => [
          ...previous,
          {
            label: parseEnrolmentServerErrorLabel({ key }),
            message: parseServerErrorMessage({ error: e as string[], t }),
          },
        ],
        []
      );
    } else {
      return [];
    }
  }
  // Get correct field name for an error item
  function parseEnrolmentServerErrorLabel({ key }: { key: string }): string {
    if (isGenericServerError(key)) {
      return '';
    }

    return t(`enrolment.form.label${pascalCase(key)}`);
  }
};

export const parseSeatsReservationServerErrors = ({
  result,
  t,
}: {
  result: LEServerError;
  t: TFunction;
}): ServerErrorItem[] => {
  return parseServerErrors({
    parseServerError: parseSeatsReservationServerError,
    result,
    t,
  });

  // Get error item for an single error.
  function parseSeatsReservationServerError({
    error,
    key,
  }: {
    error: LEServerError;
    key: string;
  }) {
    return [
      {
        label: parseSeatsReservationServerErrorLabel({ key }),
        message: parseServerErrorMessage({ error, t }),
      },
    ];
  }

  // Get correct field name for an error item
  function parseSeatsReservationServerErrorLabel({
    key,
  }: {
    key: string;
  }): string {
    if (isGenericServerError(key)) {
      return '';
    }

    return key;
  }
};
