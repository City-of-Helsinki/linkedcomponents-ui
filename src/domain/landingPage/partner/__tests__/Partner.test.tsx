import React from 'react';

import { render } from '../../../../utils/testUtils';
import Partner from '../Partner';

test('should match snapshot', () => {
  const { container } = render(
    <Partner
      href="www.testurl.com"
      imageUrl={'www.imageurl.com'}
      name="Partner name"
    />
  );

  expect(container).toMatchSnapshot();
});
