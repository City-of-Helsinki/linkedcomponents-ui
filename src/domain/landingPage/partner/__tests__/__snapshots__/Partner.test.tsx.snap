// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<div>
  <a
    aria-label="Partner name (avataan uudessa välilehdessä)"
    class="partner"
    href="www.testurl.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Partner name"
      src="www.imageurl.com"
    />
  </a>
</div>
`;
